var/spack/repos/builtin/packages/emboss/package.py

```diff
@@ -16,9 +16,19 @@ class Emboss(AutotoolsPackage):
 
     version('6.6.0', 'cc3fca80cb0618deb10fa0d29fe90e4b')
 
-    depends_on('libxpm')
+    variant('without-x', default=False, description='Build without X support')
+    variant('without-postgresql', default=False, description='Build without postgreSQL for Ensembl access library')
+
+    depends_on('libxpm', when='~without-x')
     depends_on('libgd')
-    depends_on('postgresql')
+    depends_on('postgresql', when='~without-postgresql')
+
+    def configure_args(self):
+        args = []
+        if self.spec.satisfies('+without-x'):
```

Comment:
Re the naming comment above: this is where the double negatives, etc... get confusing

Reply:
Yes, I didn't like the double-negative, but wasn't sure how it works to have a default variant (how to turn it off).  But since I made this commit, I [read up on variants](https://spack.readthedocs.io/en/latest/basic_usage.html#variants) and now I see it's not hard at all.